src/traits/data.jl

```diff
@@ -97,56 +97,145 @@ centroid(data::Data, ind) = centroid(domain(data), ind)
 # -----------------
 
 Tables.istable(::Type{<:Data}) = true
-
+Tables.columnaccess(::Type{<:Data}) = true
+Tables.columns(data::Data) = DataCols(data)
 Tables.rowaccess(::Type{<:Data}) = true
+Tables.rows(data::Data) = DataRows(DataCols(data))
+Tables.schema(data::Data) = Tables.schema(DataCols(data))
+Tables.materializer(::Type{D}) where {D<:Data} = constructor(D)
+
+# wrapper type for columns of the data table
+struct DataCols{D<:Data,C,G}
+  data::D
+  tcols::C
+  ncols::Int
+  names::Vector{Symbol}
+  domain::Vector{G}
+
+  function DataCols(data::D) where {D<:Data}
+    tcols = Tables.columns(values(data))
+    names = [Tables.columnnames(tcols)..., :geometry]
+    dom   = collect(domain(data))
+    new{D,typeof(tcols),eltype(dom)}(data, tcols, length(names), names, dom)
+  end
+end
```

Comment:
This is not the correct way of doing it @eliascarv. We just need to implement the AbstractColumns interface for DataCols. You are reimplementing the full table interface.